compiler/damlc/daml-compiler/src/DA/Daml/Compiler/DataDependencies.hs

```diff
@@ -462,6 +463,21 @@ generateSrcFromLf env = noLoc mod
             mode <- LFC.decodeOverlapMode (snd (LF.dvalBinder dval))
             Just (noLoc mode)
 
+        genInstanceBinds :: DFunSig -> Gen (LHsBinds GhcPs)
+        genInstanceBinds DFunSig{..}
+            | DFunHeadNormal{..} <- dfsHead
+            , Right (LF.TForalls _ (LF.TStruct fields)) <-
+                LF.runGamma (envWorld env) (envLfVersion env) $ do
+                    LF.expandTypeSynonyms (LF.TForalls dfsBinders (LF.TSynApp dfhName dfhArgs))
```

Comment:
Under which circumstances do we need to expand type synonyms? It seems like this definitely matters for synonyms to functions but I don’t think we have any of those do we? No objection if it’s just to be correct even if we never hit this but if we do, I’d like to know when.

Reply:
We need to get the type class definition from LF in order to generate the method stubs. The dictionary function signature only contains the type synonym and its arguments, so expanding the type synonym here gets us the underlying LF struct, and with the arguments instantiated.

Reply:
Oh, I mixed up the type we pass to `mkStubBind` where we only count function arguments (and therefore expansion shouldn’t matter) with the one here. Makes sense now :+1: